File: sprint5/By languages/C++/lesson8code4.cpp
Node* big_left_rotation(Node* v) {
    // Правым ребёнком становится новый корень правого поддерева.
    v->right = small_right_rotation(v->right);
    // Возвращаем новый корень поддерева.
    return small_left_rotation(v);
}

File: sprint5/By languages/swift/lesson8code4.swift
func bigLeftRotation<T: Comparable>(_ v: Node<T>) -> Node<T> {
    // Правым ребёнком становится новый корень правого поддерева.
    let newRight = smallRightRotation(v.right)
    v.right = newRight
    
    // Возвращаем новый корень поддерева.
    return smallLeftRotation(v)
}
